libdnf/sack/query.cpp

```diff
@@ -1321,7 +1322,7 @@ Query::Impl::filterAdvisory(const Filter & f, Map *m, int keyname)
                     eq = false;
             }
             if (eq) {
-                advisory.getPackages(pkgs, false);
+                pkgs = advisory.getPackages(false);
```

Comment:
This will not work. Anyway such changes have to be in separate commit. The previous implementation was focused on performance.